src/utils/events/eventListeners/listenerRemover.ts

```diff
@@ -0,0 +1,68 @@
+import { ClientType } from '@colony/colony-js';
+
+import { ContractEventsSignatures } from '~types';
+
+export type ListenerRemover = () => void;
+
+const removers: Record<string, ListenerRemover> = {};
+
+export const saveRemover = (
+  contractAddress: string,
+  clientType: ClientType,
+  eventSignature: ContractEventsSignatures,
+  listenerRemover: ListenerRemover,
+): void => {
+  const listenerRemoverKey = generateListenerRemoverKey(
+    contractAddress,
+    clientType,
+    eventSignature,
+  );
+  removers[listenerRemoverKey] = listenerRemover;
+};
+
+/**  Generate a unique key for the listenerRemover. */
+const generateListenerRemoverKey = (
+  contractAddress: string,
+  clientType: ClientType,
+  eventSignature: ContractEventsSignatures,
+): string => `${contractAddress}-${clientType}-${eventSignature}`;
+
+const removeListeners = (listenerRemovers: ListenerRemover[]): void => {
+  listenerRemovers.forEach((listenerRemover) => {
+    listenerRemover();
+  });
+};
+
+const getRemovers = (keys: string[]): ListenerRemover[] => {
+  const listenerRemovers: ListenerRemover[] = [];
+
+  keys.forEach((key) => {
```

Comment:
I could `map` here but this is slightly safer before it guards against the scenario where `removers[key]` returns undefined

Reply:
Very much a just in case, not expecting it to return undefined